### PR TITLE
feat: sort routes map keys alphabetically

### PR DESCRIPTION
**Pull request by zeke**
Hey @gr2m 👋 

I just took @swinton's [`github-rest-apis-for-insomnia`](https://github.com/swinton/github-rest-apis-for-insomnia) for a spin and noticed the nouns (scopes?) are not in order. This PR updates the build script to alphabetically sort the keys in the endpoints map.

I didn't commit the changes that result from running `npm run update` because I think that happens automatically on a schedule, right?

Here's the order of keys after running the script:

```sh
$ cat routes/api.github.com/index.json | npx json -k
[
  "activity",
  "apps",
  "checks",
  "codesOfConduct",
  "emojis",
  "gists",
  "git",
  "gitignore",
  "interactions",
  "issues",
  "licenses",
  "markdown",
  "meta",
  "migrations",
  "oauthAuthorizations",
  "orgs",
  "projects",
  "pulls",
  "rateLimit",
  "reactions",
  "repos",
  "scim",
  "search",
  "teams",
  "users"
]
```

(Apologies for preemptively opening a PR without first opening an issue.)

### CODE REVIEW

**Comment by zeke** [comment]
Note: this change has the side-effect of leaving the `scope` property in the resulting objects. It seems like a potentially useful property to include, but let me know if you think I should remove it.

**Comment by gr2m** [comment]
Hmmm I think the current order reflects the order at https://developer.github.com/v3/, is that possible? What’s the diff in order?

**Comment by gr2m** [comment]
you can run `bin/octokit-rest-routes.js update --cached` and `bin/octokit-rest-routes.js update --ghe  --cached` to update all route files, no need to wait for the cron

**Comment by zeke** [comment]
Here's the order of keys in the latest version of @octokit/routes:

```
$ curl -s -L "https://unpkg.com/@octokit/routes/routes/api.github.com/index.json" | json -k
[
  "oauthAuthorizations",
  "activity",
  "checks",
  "gists",
  "git",
  "apps",
  "interactions",
  "issues",
  "migrations",
  "codesOfConduct",
  "emojis",
  "gitignore",
  "licenses",
  "markdown",
  "meta",
  "rateLimit",
  "orgs",
  "projects",
  "pulls",
  "reactions",
  "repos",
  "search",
  "teams",
  "scim",
  "users"
]
```

**Comment by zeke** [comment]
Okay I ran `bin/octokit-rest-routes.js update --cached && bin/octokit-rest-routes.js update --ghe --cached`. It appears to have run successfully, but did not produce any changes.

On the other hand, `npm run update` does produce changes. Should I run that and commit the changes in this PR?

**Comment by zeke** [comment]
Getting some test failures after running `npm run update`: https://gist.github.com/zeke/9cd4ffef0e5b0f0269dd51eb9c7bd943

@gr2m can you give some guidance on how I should resolve those?

**Comment by gr2m** [comment]
> did not produce any changes

Ah, I think using the `cached` just uses the cached JSON for the route map keys instead of calculating it afresh from the cached HTML: https://github.com/octokit/routes/blob/a68368c16c9dad15f648d6448b28be9f3be5aa62/cache/api.github.com/pages.json

>  can you give some guidance on how I should resolve those?

I’m looking into it

**Comment by gr2m** [comment]
Okay the problem was that the update routes compares the separate route JSON files to look for changes. If it doesn’t find any, it says it’s all up-to-date. Because you just did a refactoring, it didn’t find any. I deleted the `routes/` folder and run the update script for api.github.com and GHE, so that works now. 

I’ll take care of the conflicts and get this merged

File: lib/check-or-update-routes.js [hunk 1]
```diff
@@ -136,17 +136,22 @@ async function checkOrUpdateRoutes (options) {
...
-      map[scopeNormalized] = []
-    }
-
-    map[scopeNormalized].push(endpoint)
 
-    return map
-  }, {})
+  const routesByScope = allEndpoints
+    .map(endpoint => {
+      endpoint.scope = camelCase(endpoint.documentationUrl.substr(state.baseUrl.length).split('/')[0])
```
Comment (gr2m) [inline lib/check-or-update-routes.js]: did you intend to add the `scope` property to the endpoint object? That’s what currently makes the tests fail. I’ll remove it for now in order to ship it, we can do a follow up PR if needed

**Comment by gr2m** [comment]
Because of the new sorting, tests are failing

```
TEST_URL=https://developer.github.com/v3/issues/#list-issues ltap test/integration/single-endpoint-test.js
```

Only endpoints are affected which include multiple routes such as https://developer.github.com/v3/issues/#list-issues

**Comment by gr2m** [comment]
I think the right place to implement the alphabetical order is in `lib/endpoint/get.js` and `lib/landing-page/get.js`, that way the tests will get the order as well as they test the library code, not the binary

**Comment by zeke** [comment]
> I think the right place to implement the alphabetical order is in lib/endpoint/get.js and lib/landing-page/get.js

Gotcha. 👍  Want me to work on that, or are you on it?

**Comment by gr2m** [comment]
I think I got it.

But I start to wonder if this is the right place to make this change. My idea was to keep routes closely aligned with the REST API docs, in order to be able to generated docs that match the current ones in structure, but have specifics for the respecitve Octokit library.

If you want to have an ordered list in Insomnia, why not sort the keys in the script that generates the set there?

**Comment by zeke** [comment]
> If you want to have an ordered list in Insomnia, why not sort the keys in the script that generates the set there?

I considered that, but figured it would make more sense to sort them here upstream because without the hierarchical context of the developer.github.com sidebar, that order (oauthAuthorizations, activity, checks, gists, git, apps... ) does not seem like a logical progression. It just seems random.

**Comment by gr2m** [comment]
So would sorting the scopes be enough, and we leave the order of the endpoints within one documentation page in tact?

**Comment by zeke** [comment]
That seems reasonable. And if it makes more sense for the Insomnia plugin to have the endpoints sorted alphabetically by name (e.g. `List Deployments`), then that change could be made on the Insomnia side instead.

**Comment by swinton** [comment]
Woah that's a big diff 😂 https://github.com/swinton/github-rest-apis-for-insomnia/pull/9/files#diff-a9ae47fead5ba8c402afea78944b56e8

**Comment by gr2m** [comment]
Sorry for the noise, that was a one-time thing, they diffs won’t usually be that big

**Comment by swinton** [comment]
> Sorry for the noise, that was a one-time thing, they diffs won’t usually be that big

No need to apologize, I was just commenting to @zeke that I am loving these changes, thanks for all the work here :bow: 💜

**Comment by zeke** [comment]
Thanks @gr2m ✨

**Pull request merged**